File: crates/kebab-parse-code/tests/fixtures/sample.java
// sample.java
package com.kebab.chunk;

import java.util.List;
import java.util.stream.Collectors;

/**
 * Heading-aware Markdown chunker.
 */
public class MdHeadingV1Chunker {
    private final String name;

    public MdHeadingV1Chunker(String name) {
        this.name = name;
    }

    public List<String> chunkDoc(String input) {
        return List.of(name, input);
    }

    public String getName() {
        return name;
    }

    public static class Builder {
        private String name;
        public Builder withName(String n) { this.name = n; return this; }
        public MdHeadingV1Chunker build() { return new MdHeadingV1Chunker(name); }
    }
}

interface Stringer {
    String asString();
}

enum Mode { DEFAULT, FAST }

File: crates/kebab-parse-code/tests/fixtures/sample.kt
// sample.kt
package com.kebab.chunk

import java.util.List

/**
 * Heading-aware Markdown chunker.
 */
class MdHeadingV1Chunker(val name: String) {
    fun chunkDoc(input: String): List<String> = listOf(name, input)

    fun getName(): String = name

    companion object {
        fun withName(n: String): MdHeadingV1Chunker = MdHeadingV1Chunker(n)
    }
}

interface Stringer {
    fun asString(): String
}

enum class Mode { DEFAULT, FAST }

fun freeFunction(x: Int): Int = x + 1

object Singleton {
    fun ping(): String = "pong"
}
